Please read the description BEFORE merging 👍

SanderM2 opened the pull request:
Hey ridiekel,

This is still a work in progress. Please wait before merging it to master.

The things changed so far:

**Dimmers:**
- They now have separate "state" and "brightness" attributes in json. This simplified integration with HA
- HA dimmers auto discovery now uses these changed attributes and can turn on lights at the previous brightness state.

**Service functions:**
- I mistakenly called them 'service functions' in a previous PR. I was in the assumption I was using function id 53, which is for the service functions. But it was id 52 which is for the digital inputs only i think. My mistake, so I renamed SERVICE functions to INPUT now.

**MQTT:**
- bugfix: After restarting the mqtt broker jeletask needs to be restarted because it's no longer subscribed to the mqtt topics.
This has been fixed in this PR by setting this.connOpts.setCleanSession(false);.
I'm unsure if we need a "clean session" at all? If we do, we need to somehow make jeletask resubscribe to the mqtt topics after a reconnect. If not, this can be considered a permanent fix.
I haven't noticed any problems with this and I'm already using setCleanSession(false) for over 2 weeks...

**AURUS sensor/wall switch:**
- In Teletask an Aurus wall switch has a temperature sensor that can be connected to a HVAC and work like a thermostat. 
I'm adding support to control this "sensor" with those SET_TEMP* functions.
So far I can send ONOFF/UP/DOWN/HEAT/COOL/... with succes. 

- The toComponentState() function is still unfinished because I'm waiting for more API details from Teletask. 
I hope they will provide. If not, I will have to continue reverse engineering the protocol...

### **ISSUE**

**Dimmers:**
Dimmers now only accept json commands on the /set topic. I'm not sure how I could have broken this?
It should still be possible to send the string instead of json. But it looks like I broke it?

PS: do we still want to accept non-json?

claesmathias commented:
Sander, do I understand correctly that I would be able to control my AC with the integrated Aurus OLED temperature sensor? Catching the MQTT messages and sending new MQTT message to my AC.

SanderM2 commented:
Yes, reading the temperature of the aurus temperature sensors already works. You can read it's temperature over MQTT and use it to do whatever you want, like controlling your AC.

claesmathias commented:
Yes reading the temperature sensor I'm already doing, but adding buttons to the Aurus and catch the events to cool/heat/hotter/colder not.

![image](https://user-images.githubusercontent.com/1142852/192722371-4c11c2ea-f898-4cc8-8186-c91a859c73be.jpeg)

ridiekel commented:
PS: do we still want to accept non-json? -> I did this because I have little time to refactor my own integration with openhab. I wanted to be able to test everything without changing that. In the long term I don't think we need this unless someone here in the forums need this. Most use HA autoconfigure, so they would not have any issues I guess.

claesmathias commented:
I believe you can remove it

SanderM2 commented:
> Yes reading the temperature sensor I'm already doing, but adding buttons to the Aurus and catch the events to cool/heat/hotter/colder not.

I think it should be possible after this PR is finished. It should then report changes like the mode changes (vent/dry/cool/heat) and changes of the set temperature….

The teletaks central reports back when changes are made but the 17 bytes it sends are undocumented. I’m still waiting for more details from Teletask. If they don’t provide we will have to reverse engineer and try to find out what each byte means. 

I only discovered the first 4 bytes (=current temperature and target temperature) + byte 12  (on/off). 

The target temperature, which is the temperature you set on the aurus, is already being reported back  now with the changes in this PR.

SanderM2 commented:
**UPDATE:**

I renamed "aurus" to "temperaturecontrol" because the support for aurus I'm adding was not specific for aurus only.
It works the same way for other devices like an AC (HVAC).

It also looks like Teletask isn't going to answer my request for more api details so I had to dig into the received bytes myself.

At this time we can already see: 
- on/off state
- current temperature
- target temperature
- current preset (DAY/NIGHT/STANDBY)
- current mode (AUTO/HEAT/COOL/VENT/DRY)
- fan speed (SPAUTO/SPLOW/SPMED/SPHIGH)

We can also control the aurus or ac by.

The following "state" attribute commands are supported:
- ON (Turn on the device)
- OFF (Turn off the device)
- ONOFF (Toggle on or off the device)
- UP (Set target temperature up 0.5°C)
- DOWN  (Set target temperature down 0.5°C)
- Presets, one of: DAY, NIGHT, STANDBY
- Mode, one of: AUTO, HEAT, COOL, VENT, DRY (or MODE to toggle between the available modes...)
- Fan speed, one of: SPAUTO, SPLOW, SPMED, SPHIGH (or SPEED to toggle between the available speeds...)

You can also set the target temperature by sending the desired temperature to the "target_temperature" attribute.

SanderM2 commented:
UPDATE:

HA autodiscovery for sensors now supported.
You can add "ha_unit_of_measurement" to your config.json for each sensor if you need "unit_of_measurement" in HA.

Regular sensors show up as a "sensor" entity. Sensors of type "TEMPERATURECONTROL" are auto discovered as a "climate" entity.

SanderM2 commented:
I have updated the readme and also removed non-json examples from it.

@ridiekel 

I'm currently running a version with all changes in this PR commited.

From HA I can now control my AC or Aurus. I get notifications on my Aurus wall switches (displaymessage) whenever someone rings the doorbell. Dimmers work fine, turning them on sets them back to the previous brightness. I can read digital inputs (garagedoor open/close for example...), my pool sensors are auto discovered in HA, my timed functions are auto discoverd and working...

Everything seems to work fine so far so I think I can call this PR finished.

claesmathias commented:
Nice! How do you glue Aurus and AC together, you implement MQTT triggers? Which Aurus screen are you using, Aurus OLED?

SanderM2 commented:
I have Aurus OLED 8 wall switches in all rooms and I have a Teletask TDS15200 license to control my AC's over the network.
So, I was already controlling my AC's from the Aurus screens (using the Teletask license.). In home assistant I control my AC's directly over the network.

If I'm not mistaken you want to read the Aurus current+target temperature/state/mode/fanspeed etc? 

These will be reported over MQTT if you add your Aurus 'sensor' like this in config.json:
```
      {
        "number": 3,
        "description": "Aurus Speelkamer",
        "type": "TEMPERATURECONTROL"
      },
```

The number of your aurus can be found in Prosoft->Diagnostics->Analog Inputs.

If It's not shown there you need to enable the Aurus temperature sensor in the aurus configuration screen and hitting the temperature icon on the left bottom of the preview screen.
